pkg/cmd/run-pgo-build/main.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+//
+// run-pgo-build triggers a run of a special roachtest job in TeamCity on the
+// current branch, waits for the job to complete, downloads the associated
+// artifacts, extracts all the CPU (.pprof) profiles, merges them, and places
+// the merged result in a location on disk.
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/pprof/profile"
+)
+
+const (
+	buildConfigID = "Cockroach_Nightlies_RoachtestGceForceProfile"
+)
+
+var (
+	branch    = os.Getenv("TC_BUILD_BRANCH")
+	serverURL = os.Getenv("TC_SERVER_URL")
+	username  = os.Getenv("TC_API_USER")
+	password  = os.Getenv("TC_API_PASSWORD")
+
+	outFile = flag.String("out", "", "where to store the result pprof profile")
+)
+
+type Build struct {
+	ID         int64
+	FinishDate string
+	Status     string
+}
+
+type ReadAtCloser interface {
+	io.ReaderAt
+	io.ReadCloser
+}
+
+func doRequest(req *http.Request) (*Build, error) {
+	req.SetBasicAuth(username, password)
+	req.Header.Add("Accept", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Do we care about the status code? Not sure if you want to proceed for statusCode >= 400.